### ISSUE TITLE
Flannel install and kube-dns CrashloopBackOff Error

### ISSUE THREAD

**Issue by iramsey85** [coreos/flannel#866]
My use case - Deploying Hypriot Kubernetes Pi cluster.

## Expected Behavior

I expect all services to be up and running with no issues

## Current Behavior

kubectl get po --all-namepaces
Namespace	Name		Ready	Running			Restarts
kube-system	kube-dns-xxxx	2/3	CrashLoopBackOff	

## Possible Solution

i'm thinking a newer version may resolve the issue or maybe I am missing a configuration.

## Steps to Reproduce (for bugs)

these commands are used to install and deploy flannel

1. $ curl -sSL https://rawgit.com/coreos/flannel/v0.7.1/Documentation/kube-flannel-rbac.yml | kubectl create -f -
2. $ curl -sSL https://rawgit.com/coreos/flannel/v0.7.1/Documentation/kube-flannel.yml | sed "s/amd64/arm/g" | kubectl create -f -

## Context

I'm trying to test the kubernetes cluster by deploying a simple service. The final step in this process is to run a curl command to browse the test app over port 80. The curl command seems to hang and timeout. This is what pointed my back to the internal pods running on the cluster.

I would also like to answer the following questions:
1. Is there a way to remove/scale down kube-dns replicas?
2. Can the flannel version be updated without tearing down the cluster?

## Your Environment
* Flannel version:v0.7.1
* Backend used (e.g. vxlan or udp):I am using the default backend, which I believe is vxlan
* Etcd version:
* Kubernetes version (if used):1.8.2
* Operating System and version:debian (hypriot image)
* Link to your project (optional):steps I followed (https://blog.hypriot.com/post/setup-kubernetes-raspberry-pi-cluster/)

**Comment by antek-drzewiecki** [comment]
+1

**Comment by tiayi** [comment]
I have got the same issue. 
Because the dns container cannot access the apiserver, other hosts, internet, any clusterIP , they just can access the host IP and other conainers (can crosss host).

I tried to debug them.  
 Container A(10.200.0.180) runs on host (172.21.1.123),  then ping another host(172..21.1.122),
tcpdmp on eth0,    can see 10.200.0.180 > 172.21.1.122  , but no response.

There is another way  can fix your DNS crush ,but pod still cannot access outside and clusterIP.
https://github.com/kubernetes/kubernetes/issues/40761
